infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -492,7 +493,13 @@ public void sealServerWithEpoch(long epoch) {
         );
 
         try {
-            batchProcessor.addTask(BatchWriterOperation.Type.SEAL, batchProcessorReq).join();
+            if(batchProcessor.getContext().getStatus() != BatchProcessorStatus.BP_STATUS_ERROR){
+                batchProcessor.addTask(BatchWriterOperation.Type.SEAL, batchProcessorReq).join();
```

Comment:
The status can change post check from `BP_STATUS_OK` to `BP_STATUS_ERROR` which would cause the join to block indefinitely.